src/blocks/weather/open_weather_map.rs

```diff
@@ -386,14 +389,25 @@ impl WeatherProvider for Service<'_> {
             })
         };
 
+        let sunrise = unix_to_datetime(current_data.sys.sunrise);
```

Comment:
I'd recommend using (with nsecs = 0): 

pub const fn from_timestamp(secs: i64, nsecs: u32) -> Option<Self>

Makes a new DateTime<Utc> from the number of non-leap seconds since January 1, 1970 0:00:00 UTC (aka “UNIX timestamp”) and the number of nanoseconds since the last whole non-leap second.